### PR TITLE
feat: update-strategy-name-and-others-fn

### PR DESCRIPTION
**Pull request by TheAustrian1998**
- add code for updateStrategyName()
- add code for _checkTokenSharesSanity()
- tests

### CODE REVIEW

**Comment by TheAustrian1998** [comment]
Thanks in advance, Sam

**Review by 0xsambugs — COMMENTED** [review]
Great work @TheAustrian1998 , really good work!

Just left some minor comments

File: contracts/DCAStrategies/DCAStrategies/DCAStrategiesManagementHandler.sol [hunk 1]
```diff
@@ -67,24 +68,42 @@ abstract contract DCAStrategiesManagementHandler is IDCAStrategiesManagementHand
...
   /// @inheritdoc IDCAStrategiesManagementHandler
-  function acceptStrategyOwnership(uint80 _strategyId) external override {}
+  function acceptStrategyOwnership(uint80 _strategyId) external {}
 
   /// @inheritdoc IDCAStrategiesManagementHandler
-  function cancelStrategyOwnershipTransfer(uint80 _strategyId) external override {}
+  function cancelStrategyOwnershipTransfer(uint80 _strategyId) external {}
+
+  function _checkTokenSharesSanity(IDCAStrategies.ShareOfToken[] memory _tokens) internal pure {
+    if (_tokens.length <= 1) revert LengthZero(); // need to have more than one item
```
[the remark this answers is not in the document]
Comment (TheAustrian1998) [reply contracts/DCAStrategies/DCAStrategies/DCAStrategiesManagementHandler.sol:97]: Agree 100%

File: test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts [hunk 2]
```diff
@@ -42,32 +44,54 @@ contract('DCAStrategiesManagementHandler', () => {
...
+    when('token shares are not equal 100%', () => {
+      let invalidShare = { token: wallet.generateRandomAddress(), share: BigNumber.from(30e2) };
+      let invalidShares = [SHARE_TOKEN_A, invalidShare];
+
+      then('tx reverted with message', async () => {
+        await expect(DCAStrategiesManagementHandlerMock.createStrategy(NAME, invalidShares, user.address)).to.be.revertedWith(
+          'InvalidTokenShares()'
+        );
+      });
+    });
```
[the remark this answers is not in the document]
Comment (TheAustrian1998) [reply test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts:76]: I tried something, it works, but it's messing up all the tests. Do you have an example I can follow?

[the remark this answers is not in the document]
Comment (0xsambugs) [reply test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts]: What do you mean that the tests get messed up?

Maybe [this example](https://github.com/Mean-Finance/dca-v2-core/blob/main/test/unit/DCAPermissionsManager/dca-permissions-manager.spec.ts#L892) helps?

[the remark this answers is not in the document]
Comment (TheAustrian1998) [reply test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts]: image
It sends the tests to the bottom and I don't like it 😅

[the remark this answers is not in the document]
Comment (0xsambugs) [reply test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts]: But why? You can call the function inside the `describe`

[the remark this answers is not in the document]
Comment (TheAustrian1998) [reply test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts]: I'm doing exactly that.
If you don't mind, I can merge what I have and try this in a later PR

**Pull request merged**